scalyr_agent/scalyr_client.py

```diff
@@ -859,10 +859,16 @@ def __init__(
         # 2->TODO use BytesIO, make all data that is going to be written here - binary
         string_buffer = io.BytesIO()
         serialized_base_body = scalyr_util.json_encode(base_body, binary=True)
-        string_buffer.write(serialized_base_body)
 
         # Now go back and find the last '}' and delete it so that we can open up the JSON again.
+        # NOTE: attributes is always a dict so we can simply rewinding logic by assuming }
+        # is always the last character when serializing a dict to json
+        string_buffer.write(serialized_base_body[:-1])
+        # Old implementation
```

Comment:
I think we should remove the old implementation from the code.

Reply:
Removed.